src/SimpleJson.ts

```diff
@@ -0,0 +1,388 @@
+		}
+
+		public WritePropertyEnd() {
+			this.Assert(this.state === SimpleJson.Writer.State.Property);
+			this.Assert(this.childCount === 1);
+			this._stateStack.pop();
+		}
+
+		// Prepare a new property name, except this time, the property name
+		// will be created by concatenating all the strings passed to
+		// WritePropertyNameInner.
+		public WritePropertyNameStart() {
+			this.Assert(this.state === SimpleJson.Writer.State.Object);
+			this.IncrementChildCount();
+
+			this._currentPropertyName = '';
+
+			this._stateStack.push(new SimpleJson.Writer.StateElement(SimpleJson.Writer.State.Property));
+			this._stateStack.push(new SimpleJson.Writer.StateElement(SimpleJson.Writer.State.PropertyName));
+		}
+
+		public WritePropertyNameEnd() {
+			this.Assert(this.state === SimpleJson.Writer.State.PropertyName);
+			this.Assert(this._currentPropertyName !== null);
+			this._propertyNameStack.push(this._currentPropertyName!);
+			this._currentPropertyName = null;
+			this._stateStack.pop();
+		}
+
+		public WritePropertyNameInner(str: string) {
+			this.Assert(this.state === SimpleJson.Writer.State.PropertyName);
+			this.Assert(this._currentPropertyName !== null);
+			this._currentPropertyName += str;
+		}
+
+		// Add a new array.
+		public WriteArrayStart() {
+			this.StartNewObject(true);
+
+			let newObject: any[] = [];
+
+			if (this.state === SimpleJson.Writer.State.Property) {
+				// This array is created as the value of a property,
+				// inside an object.
+				this.Assert(this.currentCollection !== null);
+				this.Assert(this.currentPropertyName !== null);
+
+				let propertyName = this._propertyNameStack.pop();
+				this.currentCollection![propertyName!] = newObject;
+				this._collectionStack.push(newObject);
+			} else if (this.state === SimpleJson.Writer.State.Array) {
+				// This array is created as the child of another array.
+				this.Assert(this.currentCollection !== null);
+
+				this.currentCollection!.push(newObject);
+				this._collectionStack.push(newObject);
+			} else {
+				// This array is the root object.
+				this.Assert(this.state === SimpleJson.Writer.State.None);
+				this._jsonObject = newObject;
+				this._collectionStack.push(newObject);
+			}
+
+			this._stateStack.push(new SimpleJson.Writer.StateElement(SimpleJson.Writer.State.Array));
+		}
+
+		public WriteArrayEnd() {
+			this.Assert(this.state === SimpleJson.Writer.State.Array);
+			this._collectionStack.pop();
+			this._stateStack.pop();
+		}
+
+		// Add the value to the appropriate collection, given the current
+		// context.
+		public Write(value: number | string | boolean | null, escape: boolean = true) {
+			if (value === null) {
+				console.error('Warning: trying to write a null string');
+				return;
+			}
+
+			this.StartNewObject(false);
+			this._addToCurrentObject(value);
+		}
+
+		public WriteInt(value: number) {
+			this.StartNewObject(false);
+			this._addToCurrentObject(Math.floor(value));
```

Comment:
Wouldn't `parseInt` first convert to a string, then back again to a number?

Reply:
Mmmh maybe internally, but `parseInt(<Number>)` is still a valid call AFAIK. Like i said my main concern is that I had to refer to the method name to figure out why we are flooring the number. A comment to clarify would work too!

Reply:
Actually, TypeScript won't let me use `parseInt` because it expects a `string`. I'd have to make the conversion beforehand, which would look weirder I think. I'll add a comment to clarify!

I looked through the codebase and `parseInt` is mostly used when deserialising stuff; since we can't make assumptions about the type (it's marked as `any`), it makes sense.